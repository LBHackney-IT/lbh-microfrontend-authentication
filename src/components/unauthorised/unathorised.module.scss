@import "~lbh-frontend/lbh/base";
@import "~lbh-frontend/lbh/core/all";
@import "~lbh-frontend/lbh/objects/all";

.unauthorised {
  border: 1px solid lbh-colour("lbh-a04");
  border-left-width: 15px;
  padding: 30px;
  background: lighten(lbh-colour("lbh-a04"), 20%);

  @include lbh-font;

  p {
    margin: 0;
    padding: 9px;
    font-weight: bold;
    font-size: 2rem;
    line-height: 1.2rem;
  }
}
