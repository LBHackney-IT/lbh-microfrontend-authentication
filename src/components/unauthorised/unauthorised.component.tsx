import React from 'react';

import './unathorised.scss';
import { Button, ErrorSummary } from '@mtfh/common';
import { locale } from '../../services';
import { REQUEST_ACCESS_LINK } from '../../constants';

export const Unauthorised = (): JSX.Element => {
    return (
        <div
            className="govuk-error-summary lbh-error-summary"
            aria-labelledby="error-summary-title"
            role="alert"
            tabIndex={-1}
            data-module="govuk-error-summary"
        >
            <ErrorSummary
                id="error-summary-title"
                title={locale.unauthorisedToViewService}
            />
            <div className="govuk-error-summary__body">
                <p>
                    {locale.contactToRequestPermission}
                    <Button as="a" href={REQUEST_ACCESS_LINK}>
                        {locale.requestAccess}
                    </Button>
                </p>
            </div>
        </div>
    );
};
