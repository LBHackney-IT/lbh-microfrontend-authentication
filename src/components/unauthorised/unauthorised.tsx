import React from "react";

import { REQUEST_ACCESS_LINK } from "../../constants";
import { locale } from "../../services";

import { Button, ErrorSummary } from "@mtfh/common/lib/components";
import "./unathorised.scss";

export const Unauthorised = (): JSX.Element => {
  return (
    <ErrorSummary id="error-summary-title" title={locale.unauthorisedToViewService}>
      <p>
        {locale.contactToRequestPermission}
        <Button as="a" href={REQUEST_ACCESS_LINK}>
          {locale.requestAccess}
        </Button>
      </p>
    </ErrorSummary>
  );
};
