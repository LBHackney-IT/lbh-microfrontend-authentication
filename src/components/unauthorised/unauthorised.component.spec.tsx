import React from 'react';
import { render } from '@testing-library/react';

import { Unauthorised } from './unauthorised.component';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

it('renders correctly', async () => {
    const { container } = render(<Unauthorised />);
    expect(container).toMatchSnapshot();
});
