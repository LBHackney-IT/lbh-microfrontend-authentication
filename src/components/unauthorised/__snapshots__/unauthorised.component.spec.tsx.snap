// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    aria-labelledby="error-summary-title"
    class="govuk-error-summary lbh-error-summary"
    data-module="govuk-error-summary"
    role="alert"
    tabindex="-1"
  >
    <div
      aria-labelledby="error-summary-title"
      class="govuk-error-summary lbh-error-summary"
      role="alert"
    >
      <h2
        class="govuk-error-summary__title"
        id="error-summary-title"
      >
        You do not have permission to access this service.
      </h2>
    </div>
    <div
      class="govuk-error-summary__body"
    >
      <p>
        To request access please complete this form
        <a
          class="govuk-button lbh-button"
          href="https://docs.google.com/forms/d/e/1FAIpQLSdY3SZtpFhwgKG_P1ad6mLH-Jk3vVyFraSmbMzN_mVcciSC6w/viewform"
        >
          Request access
        </a>
      </p>
    </div>
  </div>
</div>
`;
